pkg/controller/garbagecollector/operations.go

```diff
@@ -34,9 +34,14 @@ import (
 // namespace> tuple to a unversioned.APIResource struct.
 func (gc *GarbageCollector) apiResource(apiVersion, kind string, namespaced bool) (*metav1.APIResource, error) {
 	fqKind := schema.FromAPIVersionAndKind(apiVersion, kind)
-	mapping, err := gc.restMapper.RESTMapping(fqKind.GroupKind(), apiVersion)
+	mapping, err := gc.dynamicMapper.RESTMapping(fqKind.GroupKind(), apiVersion)
 	if err != nil {
-		return nil, fmt.Errorf("unable to get REST mapping for kind: %s, version: %s", kind, apiVersion)
+		// reset the mapper so that next call will communicate with the server to re-discover.
+		gc.dynamicMapper.Reset()
+		mapping, err = gc.dynamicMapper.RESTMapping(fqKind.GroupKind(), apiVersion)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get REST mapping for kind: %s, version: %s", kind, apiVersion)
```

Comment:
The watchable discovery will solve the problem cleanly but that's not going to happen in this release.